### DISCUSSION TITLE
Using beforeInteractive strategy outside _document level.

### DISCUSSION THREAD

**Discussion by straube** [vercel/next.js#37545] [Help] [answered]
On Next.js version 12.1.6, if I have a script using `beforeInteractive` strategy outside a custom `_document`, a warning is shown. I was reading through the [related PR](https://github.com/vercel/next.js/pull/31936) and it looks like a new strategy was added for these cases: `beforePageRender`. However, it's not documented [here](https://nextjs.org/docs/basic-features/script) nor [here](https://nextjs.org/docs/api-reference/next/script). Is this a beta/unstable feature?

**Comment by icyJoseph** [comment, marked as answer]
It's not out yet. In fact, there's a bug on `beforeInteractive`, which makes it so that it is not working 😢  - It has been fixed and there's more to come in upcoming releases. 

The last stable release 12.1.6 was back on May 2nd, and a lot has been happening. The types don't really allow `beforePageRender`.

More relevant links:

- https://github.com/vercel/next.js/pull/37000
- https://github.com/vercel/next.js/issues/37225#issuecomment-1142390675
- https://github.com/vercel/next.js/pull/37084,
